src/interpreter/internal.rs

```diff
@@ -166,10 +166,41 @@ impl<S> Interpreter<S> {
         }?;
 
         // update the local copy of the output
-        let mut output = Output::variable(owner_to_set, amount_to_set, asset_id_to_update);
+        let output = Output::variable(owner_to_set, amount_to_set, asset_id_to_update);
+        self.set_output(out_idx, output)
+    }
+
+    pub(crate) fn revert_variable_output(&mut self, out_idx: usize) -> Result<(), RuntimeError> {
```

Comment:
This function is called only once `src/interpreter/post_execution.rs`.

Do we really need to generalize this behavior? As a result, we are performing the `out_idx` check more than once.